esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -3617,24 +3617,30 @@ void CWsDfuEx::setDFUQueryFilters(IEspDFUQueryRequest& req, StringBuffer& filter
     if (!req.getIncludeSuperOwner_isNull() && req.getIncludeSuperOwner())
         filterBuf.append(DFUQFTincludeFileAttr).append(DFUQFilterSeparator).append(DFUQSFAOincludeSuperOwner).append(DFUQFilterSeparator);
 
-    __int64 sizeFrom = req.getFileSizeFrom();
-    __int64 sizeTo = req.getFileSizeTo();
-    if ((sizeFrom > 0) || (sizeTo > 0))
-    {
-        StringBuffer buf;
-        if (sizeFrom > 0)
-            buf.append(sizeFrom);
-        buf.append(DFUQFilterSeparator);
-        if (sizeTo > 0)
-            buf.append(sizeTo);
-        filterBuf.append(DFUQFTinteger64Range).append(DFUQFilterSeparator).append(getDFUQFilterFieldName(DFUQFFattrsize));
-        filterBuf.append(DFUQFilterSeparator).append(buf.str()).append(DFUQFilterSeparator);
-    }
+    setInt64RangeFilter(req.getFileSizeFrom(), req.getFileSizeTo(), DFUQFFattrsize, filterBuf);
+    setInt64RangeFilter(req.getMaxSkewFrom(), req.getMaxSkewTo(), DFUQFFattrmaxskew, filterBuf);
+    setInt64RangeFilter(req.getMinSkewFrom(), req.getMinSkewTo(), DFUQFFattrminskew, filterBuf);
 
     setTimeRangeFilter(req.getStartDate(), req.getEndDate(), DFUQFFtimemodified, filterBuf);
     setTimeRangeFilter(req.getStartAccessedTime(), req.getEndAccessedTime(), DFUQFFaccessed, filterBuf);
 }
 
+void CWsDfuEx::setInt64RangeFilter(__int64 from, __int64 to, DFUQFilterField filterID, StringBuffer &filterBuf)
+{
+    if ((from == 0) && (to == 0))
+        return;
+
+    filterBuf.append(DFUQFTinteger64Range).append(DFUQFilterSeparator);
+    filterBuf.append(getDFUQFilterFieldName(filterID)).append(DFUQFilterSeparator);
+    StringBuffer buf;
+    if (from > 0)
+        filterBuf.append(from);
```

Comment:
question: I assume it is correct/ok, and not new, but is it ok when from/to are missing, that it means the filterBuf will container multiple DFUQFilterSeparator without any value between them?
Would it be better to populate with a low value (0) for from, and a high value (max-int) for to ?

Reply:
In CIterateFileFilterContainer (dadfs.cpp), there is the code to handle the multiple DFUQFilterSeparator without any value between them. The code uses the hasFilter and hasFilterHigh in checkInteger64RangeFilter() and checkIntegerRangeFilter().